tests/components/rest/test_switch.py

```diff
@@ -16,254 +16,239 @@
-        asyncio.run_coroutine_threadsafe(
-            self.switch.async_update(), self.hass.loop
-        ).result()
-
-        assert self.switch.is_on
-
-    def test_update_when_off(self, aioclient_mock):
-        """Test update when switch is off."""
-        aioclient_mock.get(self.resource, text=self.body_off.template)
-        asyncio.run_coroutine_threadsafe(
-            self.switch.async_update(), self.hass.loop
-        ).result()
-
-        assert not self.switch.is_on
-
-    def test_update_when_unknown(self, aioclient_mock):
-        """Test update when unknown status returned."""
-        aioclient_mock.get(self.resource, text="unknown status")
-        asyncio.run_coroutine_threadsafe(
-            self.switch.async_update(), self.hass.loop
-        ).result()
-
-        assert self.switch.is_on is None
-
-    def test_update_timeout(self, aioclient_mock):
-        """Test update when timeout occurs."""
-        aioclient_mock.get(self.resource, exc=asyncio.TimeoutError())
-        asyncio.run_coroutine_threadsafe(
-            self.switch.async_update(), self.hass.loop
-        ).result()
-
-        assert self.switch.is_on is None
+    assert aioclient_mock.call_count == 1
+
+
+async def test_setup(hass, aioclient_mock):
+    """Test setup with valid configuration."""
+    aioclient_mock.get("http://localhost", status=HTTP_OK)
+    assert await async_setup_component(
+        hass,
+        SWITCH_DOMAIN,
+        {
+            SWITCH_DOMAIN: {
+                CONF_PLATFORM: rest.DOMAIN,
+                CONF_NAME: "foo",
+                CONF_RESOURCE: "http://localhost",
+                CONF_HEADERS: {"Content-type": CONTENT_TYPE_JSON},
+                rest.CONF_BODY_ON: "custom on text",
+                rest.CONF_BODY_OFF: "custom off text",
+            }
+        },
+    )
+    assert aioclient_mock.call_count == 1
+    assert_setup_component(1, SWITCH_DOMAIN)
+
+
+async def test_setup_with_state_resource(hass, aioclient_mock):
+    """Test setup with valid configuration."""
+    aioclient_mock.get("http://localhost", status=HTTP_NOT_FOUND)
+    aioclient_mock.get("http://localhost/state", status=HTTP_OK)
+    assert await async_setup_component(
+        hass,
+        SWITCH_DOMAIN,
+        {
+            SWITCH_DOMAIN: {
+                CONF_PLATFORM: rest.DOMAIN,
+                CONF_NAME: "foo",
+                CONF_RESOURCE: "http://localhost",
+                rest.CONF_STATE_RESOURCE: "http://localhost/state",
+                CONF_HEADERS: {"Content-type": CONTENT_TYPE_JSON},
+                rest.CONF_BODY_ON: "custom on text",
+                rest.CONF_BODY_OFF: "custom off text",
+            }
+        },
+    )
+    assert aioclient_mock.call_count == 1
+    assert_setup_component(1, SWITCH_DOMAIN)
+
+
+"""Tests for REST switch platform."""
+
+name = "foo"
+method = "post"
+resource = "http://localhost/"
+state_resource = resource
+headers = {"Content-type": CONTENT_TYPE_JSON}
+auth = None
```

Comment:
Please make constants UPPERCASE and move to the top of the file.

Reply:
Done